### PR TITLE
Doc update

### PR DESCRIPTION
**Pull request by yuokamoto**
build result
https://rclue.readthedocs.io/en/doc_update/index.html

### CODE REVIEW

File: docs/source/examples.rst [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-Setup and run
----------------
-1. Download UE5.1 for Linux by following [Unreal Engine for Linux](https://www.unrealengine.com/en-US/linux)
+
+.. _setup_and_run_ue_project:
+
+^^^^^^^^^^^^^^^^^^^^^^^^
+Setup and run UE Project
+^^^^^^^^^^^^^^^^^^^^^^^^
+1. Download UE5.1 for Linux by following `Unreal Engine for Linux <https://www.unrealengine.com/en-US/linux>`_.
```
Comment (Tadinu) [inline docs/source/examples.rst]:
```suggestion
1. Download the lastest UE5 for Linux by following `Unreal Engine for Linux <https://www.unrealengine.com/en-US/linux>`_.
```

**Review by Tadinu — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -22,15 +24,17 @@ Please download UE5.1 for Linux by following [Unreal Engine for Linux](https://w
 ## Branches
 
 - `devel`: This build of the plugin is based on ROS2 Foxy and has been tested on Ubuntu 20 and UE5.1.
-- `UE5_devel_humble_20.04`(experimental):  This build of the plugin is based on ROS2 humble and has been tested on Ubuntu 20.04 and UE5.10.
-- `UE5_devel_humble`(experimental):  This build of the plugin is based on ROS2 humble, Ubuntu 22.04 and UE5.1.
+- `UE5_devel_humble_20.04`(experimental): This build of the plugin is based on ROS2 humble and has been tested on Ubuntu 20.04 and UE5.10.
```
Comment (larryng-rapyuta) [inline README.md]: Isnt it 5.1 or 5.1.1?
```suggestion
- `UE5_devel_humble_20.04`(experimental): This build of the plugin is based on ROS2 humble and has been tested on Ubuntu 20.04 and UE5.1.
```

File: docs/source/design.rst [hunk 3]
```diff
@@ -65,18 +65,23 @@ with other components (services, actions) following the same principles.
...
+`ROS2Node <doxygen_generated/html/d7/d68/class_u_r_o_s2_node_component.html>`_
+is implemented as UE ActorComponent, the base class for objects 
+that can be attached to the UE `Actor <https://docs.unrealengine.com/5.1/en-US/actors-in-unreal-engine/>`_ 
+such as Robot. 
+The ROS2Node tracks and manages publishers, subscribers, service clients and servers, action clients and servers 
+(and thus contains a simplified equivalent of rclcpp’s executor).
+
+ROS2 Publisher(Service/Action Client)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+`ROS2Pulisher <doxygen_generated/html/d6/dd4/class_u_r_o_s2_publisher.html>`_
```
Comment (tetorea) [inline docs/source/design.rst]:
```suggestion
`ROS2Publisher <doxygen_generated/html/d6/dd4/class_u_r_o_s2_publisher.html>`_
```

File: docs/source/examples/publisher_example.rst [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+        FROSStr msg;
+        msg.Data = FString::Printf(TEXT("%s %d"), *Message, Count++);
+        CastChecked<UROS2StrMsg>(InMessage)->SetMsg(msg);
+    }
+
+
+.. _publisher_examin_code:
+
+^^^^^^^^^^^^^^^^^^
+Examin the code
```
Comment (tetorea) [inline docs/source/examples/publisher_example.rst]:
```suggestion
Code explanations
```

File: docs/source/examples/publisher_example.rst [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+
+.. code-block:: C++
+
+    void AROS2PublisherNode::BeginPlay()
+    {
+        Super::BeginPlay();
+
+        Node->Init();
+
+then starts adding publishers. We intorduce 3 ways to add publisher here.
```
Comment (tetorea) [inline docs/source/examples/publisher_example.rst]:
```suggestion
then starts adding publishers. We introduce 3 ways to add publisher here.
```

File: docs/source/examples/publisher_example.rst [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+
+-----------------------------
+BP Publisher
+-----------------------------
+
+Blueprint implementation of a publisher is very similar to a C++ implementation. 
+Blueprints allow you to set logic/processes, parameters, and other details from the editor.
+
+You can add component such as UROS2Publisher from `Components` panel in the editor(left side in the fig below)
+and set each component parameters in `Detales` panel in the editor(right side in the fig below).
```
Comment (tetorea) [inline docs/source/examples/publisher_example.rst]:
```suggestion
and set each component parameters in `Details` panel in the editor(right side in the fig below).
```

File: docs/source/examples/publisher_example.rst [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+The main difference from the C++ implementation is that it uses 
+`UROS2PublisherComponent <../doxygen_generated/html/db/d27/class_u_r_o_s2_publisher_component.html>`_
+instead of UROS2Publisher. 
+As UROS2PublisherComponent is a child class of `UActorComponent <https://docs.unrealengine.com/5.1/en-US/API/Runtime/Engine/Components/UActorComponent/>`_
+and has UROS2Publisher as a member variable, you can easily add it to the Actor and set parameters from the editor.
+
+Each publisher is attached to an Actor, which is displayed in the `Components`` panel on the left.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Detailes of BP Publisher
```
Comment (tetorea) [inline docs/source/examples/publisher_example.rst]:
```suggestion
Details of BP Publisher
```

File: docs/source/examples/service_client_example.rst [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+
+        FROSAddTwoIntsRes res;
+        AddTwoIntsService->GetResponse(res);
+
+        // Log response
+        UE_LOG_WITH_INFO_NAMED(LogTurtlebot3, Log, TEXT("[%s][C++][receive response] sum:%d"), *ServiceName, res.Sum);
+    }
+
+^^^^^^^^^^^^^^^^^^
+Examin the code
```
Comment (tetorea) [inline docs/source/examples/service_client_example.rst:124]:
```suggestion
Code explanations
```

File: docs/source/examples/service_client_example.rst [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+    #define ROS2_CREATE_SERVICE_CLIENT_WITH_QOS(                                                  \
+        InROS2Node, InUserObject, InServiceName, InSrvClass, InRequestCallback, InQoS, OutClient) \
+        if (ensure(IsValid(InROS2Node)))                                                          \
+        {                                                                                         \
+            FServiceCallback req;                                                                 \
+            req.BindDynamic(InUserObject, InRequestCallback);                                     \
+            OutClient = InROS2Node->CreateServiceClient(InServiceName, InSrvClass, req, InQoS);   \
+        }
+
+In this example, service client is set to send request periodically by create
```
Comment (tetorea) [inline docs/source/examples/service_client_example.rst:191]:
```suggestion
In this example, service client is set to send request periodically by creating a 
```

File: docs/source/examples/service_server_example.rst [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+        // Log request and response
+        UE_LOG_WITH_INFO_NAMED(
+            LogTurtlebot3, Log, TEXT("[%s][C++][receive request] %d + %d = %d"), *ServiceName, req.A, req.B, res.Sum);
+    }
+
+
+
+^^^^^^^^^^^^^^^^^^
+Examin the code
```
Comment (tetorea) [inline docs/source/examples/service_server_example.rst]:
```suggestion
Code explanations
```

File: docs/source/examples/service_server_example.rst [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+^^^^^^^^^^^^^^^^^^
+
+On an AROS2ServiceServerNode Actor, similar to the AROS2PublisherrNode,
```
Comment (tetorea) [inline docs/source/examples/service_server_example.rst]:
```suggestion
On an AROS2ServiceServerNode Actor, similar to the AROS2PublisherNode, 
```

File: docs/source/examples/subscriber_example.rst [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+        {
+            FROSStr msg;
+            stringMsg->GetMsg(msg);
+            UE_LOG_WITH_INFO_NAMED(LogTurtlebot3, Log, TEXT("%s"), *msg.Data);
+        }
+    }
+
+
+^^^^^^^^^^^^^^^^^^
+Examin the code
```
Comment (tetorea) [inline docs/source/examples/subscriber_example.rst]:
```suggestion
Code explanations
```

File: docs/source/examples/subscriber_example.rst [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+^^^^^^^^^^^^^^^^^^
+
+On an AROS2SubscriberNode Actor, similar to the AROS2PublisherrNode,
```
Comment (tetorea) [inline docs/source/examples/subscriber_example.rst]:
```suggestion
On an AROS2SubscriberNode Actor, similar to the AROS2PublisherNode, 
```

File: docs/source/examples/subscriber_example.rst [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+
+-----------------------------
+BP Subscriber
+-----------------------------
+
+Blueprint implementation of a subscriber is very similar to a C++ implementation. 
+Blueprints allow you to set logic/processes, parameters, and other details from the editor.
+
+You can add component such as UROS2Publisher from `Components` panel in the editor(left side in the fig below)
+and set each component parameters in `Detals` panel in the editor(right side in the fig below).
```
Comment (tetorea) [inline docs/source/examples/subscriber_example.rst]:
```suggestion
and set each component parameters in `Details` panel in the editor (right side in the fig below).
```

**Review by tetorea — APPROVED** [review]

File: docs/source/examples/publisher_example.rst [hunk 15]
```diff
@@ -0,0 +1,378 @@
...
+        Node->Namespace = TEXT("cpp");
+    }
+
+    void AROS2PublisherNode::BeginPlay()
+    {
+        Super::BeginPlay();
+
+        Node->Init();
+
+        // Create publisher with 3 different way.
```
Comment (Antho1426RR) [inline docs/source/examples/publisher_example.rst]:
```suggestion
        // Create publisher with 3 different ways.
```

File: docs/source/examples/publisher_example.rst [hunk 16]
```diff
@@ -0,0 +1,378 @@
...
+and its parameters such as Node Name and Namespace are set.
+
+Although the UROS2NodeComponent is created, 
+the ROS2 Node is not yet initialized at this point. 
+Initialization of the ROS2 Node occurs during the 
+`BeginPlay <https://docs.unrealengine.com/5.1/en-US/API/Runtime/Engine/GameFramework/AActor/BeginPlay/>`_ 
+method, which is called when the simulation starts.
+
+One reason to initialize in BeginPlay is to create a ROS2 Node when the simulation starts. 
+It is important to note that the UE Class constructor may also be called before simulation start, for example,
```
Comment (Antho1426RR) [inline docs/source/examples/publisher_example.rst]:
```suggestion
It is important to note that the UE Class constructor may also be called before simulation starts, for example, 
```

File: docs/source/examples/publisher_example.rst [hunk 17]
```diff
@@ -0,0 +1,378 @@
...
+Similar to the C++ implementation, the BP class has three types of publishers: 
+Non-Loop Publisher, Loop Publisher, and Custom Publisher.
+
+The main difference from the C++ implementation is that it uses 
+`UROS2PublisherComponent <../doxygen_generated/html/db/d27/class_u_r_o_s2_publisher_component.html>`_
+instead of UROS2Publisher. 
+As UROS2PublisherComponent is a child class of `UActorComponent <https://docs.unrealengine.com/5.1/en-US/API/Runtime/Engine/Components/UActorComponent/>`_
+and has UROS2Publisher as a member variable, you can easily add it to the Actor and set parameters from the editor.
+
+Each publisher is attached to an Actor, which is displayed in the `Components`` panel on the left.
```
Comment (Antho1426RR) [inline docs/source/examples/publisher_example.rst]:
```suggestion
Each publisher is attached to an Actor, which is displayed in the `Components` panel on the left.
```

File: docs/source/examples/publisher_example.rst [hunk 18]
```diff
@@ -0,0 +1,378 @@
...
+.. image:: ../images/publisher_non_loop.png
+
+Compared to C++, which uses CreatePublisher(), 
+in Blueprint, the Publisher is already generated as a Component before BeginPlay. 
+Therefore, we use 
+`UROS2NodeComponent::AddPublisher <../doxygen_generated/html/d7/d68/class_u_r_o_s2_node_component.html#a582299af64efaaa34b046c00b1c96828>_` 
+to initialize the Publisher instead. 
+The CreatePublisher function in C++ internally calls AddPublisher.
+
+You can see that parameters such as Topic Name, Publisher Class, and Msg Class are set in the right `Details`` panel.
```
Comment (Antho1426RR) [inline docs/source/examples/publisher_example.rst]:
```suggestion
You can see that parameters such as Topic Name, Publisher Class, and Msg Class are set in the right `Details` panel.
```

File: docs/source/examples/service_client_example.rst [hunk 19]
```diff
@@ -0,0 +1,296 @@
...
+=============================
+
+Please follow the instructions in  :ref:`setup_and_run_ue_project` to setup the UE project 
+and open  `ROS2ServiceExample.umap <https://github.com/rapyuta-robotics/turtlebot3-UE/blob/devel/Content/Maps/ROS2TopicExamples.umap>`_.
+
+-----------------------------
+C++ Service Client
+-----------------------------
+
+This example send service request every 1 second.
```
Comment (Antho1426RR) [inline docs/source/examples/service_client_example.rst]:
```suggestion
This example sends service request every 1 second.
```

File: docs/source/examples/service_client_example.rst [hunk 20]
```diff
@@ -0,0 +1,296 @@
...
+                                        ServiceName,
+                                        UROS2AddTwoIntsSrv::StaticClass(),
+                                        &AROS2ServiceClientNode::ReceiveResponse,    // Callback for response
+                                        UROS2QoS::DynamicBroadcaster,
+                                        AddTwoIntsSrvClient)
+                                        
+The implementation of ROS2_CREATE_SERVICE_CLIENT_WITH_QOS is as follows. 
+It uses Unreal Engine's dynamic delegate to call the bound function 
+when the node receives the message. 
+You can find more information about Unreal Engine's dynamic delegate .
```
Comment (Antho1426RR) [inline docs/source/examples/service_client_example.rst:175]:
```suggestion
You can find more information about Unreal Engine's dynamic delegate
```

File: docs/source/examples/service_client_example.rst [hunk 21]
```diff
@@ -0,0 +1,296 @@
...
+
+
+-----------------------------
+BP Service Client
+-----------------------------
+
+Blueprint implementation of a service client is very similar to a C++ implementation. 
+Blueprints allow you to set logic/processes, parameters, and other details from the editor.
+
+You can add component such as UROS2Publisher from `Components` panel in the editor(left side in the fig below)
```
Comment (Antho1426RR) [inline docs/source/examples/service_client_example.rst]:
```suggestion
You can add components such as UROS2Publisher from `Components` panel in the editor (left side in the fig below)
```

File: docs/source/examples/service_client_example.rst [hunk 22]
```diff
@@ -0,0 +1,296 @@
...
+and set each component parameters in `Details` panel in the editor(right side in the fig below).
```
Comment (Antho1426RR) [inline docs/source/examples/service_client_example.rst]:
```suggestion
and set each component parameters in `Details` panel in the editor (right side in the fig below).
```

File: docs/source/examples/service_client_example.rst [hunk 23]
```diff
@@ -0,0 +1,296 @@
...
+We also create a UE Timer to send request every 1 second.
+
+.. image:: ../images/service_client_req.png
+
+In the timer event, SendRequest is called to send request.
+Then increment the value of A and B and print those.
+
+.. image:: ../images/service_client_res.png
+
+Callback function is bound to a custom event, indicated by the red node in the left.
```
Comment (Antho1426RR) [inline docs/source/examples/service_client_example.rst]:
```suggestion
Callback function is bound to a custom event, indicated by the red node on the left. 
```

File: docs/source/examples/service_server_example.rst [hunk 24]
```diff
@@ -0,0 +1,222 @@
...
+
+        // Log request and response
+        UE_LOG_WITH_INFO_NAMED(
+            LogTurtlebot3, Log, TEXT("[%s][C++][receive request] %d + %d = %d"), *ServiceName, req.A, req.B, res.Sum);
+    }
+
+The implementation of ROS2_CREATE_SERVICE_SERVER is as follows. 
+It uses Unreal Engine's dynamic delegate to call the bound function 
+when the node receives the message. 
+You can find more information about Unreal Engine's dynamic delegate .
```
Comment (Antho1426RR) [inline docs/source/examples/service_server_example.rst]:
```suggestion
You can find more information about Unreal Engine's dynamic delegate
```

File: docs/source/examples/service_server_example.rst [hunk 25]
```diff
@@ -0,0 +1,222 @@
...
+Therefore, we use 
+`UROS2NodeComponent::AddServiceServer <../doxygen_generated/html/d7/d68/class_u_r_o_s2_node_component.html#a88c4ddd3037e78b68f62c142015ea91e>`_
+to initialize the UROS2ServiceServer and 
+`UROS2ServiceServer::SetDelegates <../doxygen_generated/html/d8/d6e/class_u_r_o_s2_service_server.html#a3071892b4f131d18e9f5010077cad485>`_ 
+to bind callback method instead. 
+The ROS2_CREATE_SERVICE_SERVER macro in C++ internally calls CreateServiceServer which calls AddServiceServer and SetDelegates.
+
+.. image:: ../images/service_server_req.png
+
+Callback function is bound to a custom event, indicated by the red node in the left.
```
Comment (Antho1426RR) [inline docs/source/examples/service_server_example.rst]:
```suggestion
Callback function is bound to a custom event, indicated by the red node on the left. 
```

**Review by Antho1426RR — APPROVED** [review]

**Review by larryng-rapyuta — APPROVED** [review]

**Pull request merged**